public/app/plugins/datasource/loki/queryUtils.ts

```diff
@@ -293,3 +293,56 @@ export const getLokiQueryFromDataQuery = (query?: DataQuery): LokiQuery | undefi
 
   return query;
 };
+
+export const sortLabelSelectors = (query: LokiQuery): LokiQuery => {
+  const selectorPositions = getStreamSelectorPositions(query.expr);
+
+  const selectors = selectorPositions.map((selectorPos) => {
+    return query.expr.substring(selectorPos.from, selectorPos.to);
+  });
+
+  const selectorMap = selectors.map((selector) => {
+    const matcherNodes = getNodesFromQuery(selector, [Matcher]);
+
+    matcherNodes.sort((a, b) => {
+      const labelNodeA = a.getChild(Identifier);
+      const labelNodeB = b.getChild(Identifier);
+      const labelNameA = labelNodeA && selector.substring(labelNodeA.from, labelNodeA.to);
+      const labelNameB = labelNodeB && selector.substring(labelNodeB.from, labelNodeB.to);
+
+      if (!labelNameA || !labelNameB) {
+        return 0;
+      }
+
+      if (labelNameA < labelNameB) {
+        return -1;
+      }
+
+      if (labelNameA > labelNameB) {
+        return 1;
+      }
+
+      return 0;
```

Comment:
We could shorten this a little bit like:

```suggestion
      if (!labelNameA || !labelNameB) {
        return 0;
      }

      if (labelNameA < labelNameB) {
        return -1;
      }

      return (labelNameA > labelNameB) ? 1 : 0;
```